src/Validator/ResponseValidator.php

```diff
@@ -75,4 +90,18 @@ public function assertScalarResultWithId(ResponseInterface $response, string $ty
             throw ResponseValidationException::scalarResultExpected($response, $response->document()->data()->count());
         }
     }
+
+    /**
+     * @throws ResponseValidationException
+     */
+    public function assertScalarResultWithoutId(ResponseInterface $response, string $type): void
```

Comment:
A test is missing and I guess it should validate, that there is no id in the resource?

Reply:
I prefer the optional id. There is no need to force, not to have an id. An submitted id will not change the result, because no id is expected.

The method naming could be changed to `assertScalarResultWithOptionalId`.